ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -239,6 +240,42 @@ private KsqlSerdeFactory getValueSerdeFactory(final Sink sink) {
       return serdeFactories.create(format, sink.getProperties());
     }
 
+    /**
+     * Get the list of select expressions that are <i>not</i> for meta and key fields in the source
+     * schema.
+     *
+     * <p>Currently, the select expressions can include metadata and key fields, which are latter
+     * removed by {@link io.confluent.ksql.planner.plan.KsqlStructuredDataOutputNode}. When building
+     * the {@link SerdeOptions} its important the finally schema is used, i.e. the one with these
+     * meta and key fields removed.
+     *
+     * <p>This is weird functionality, but maintained for backwards compatibility for now.
+     *
+     * @return the list of field names in the sink that are not meta or key fields.
+     */
+    private List<String> getNoneMetaOrKeySelectAliases() {
+      final SourceSchemas sourceSchemas = analysis.getFromSourceSchemas();
+      final List<Expression> selects = analysis.getSelectExpressions();
+
+      final List<String> columnNames = new ArrayList<>(analysis.getSelectExpressionAlias());
+
+      for (int idx = selects.size() - 1; idx >= 0; --idx) {
```

Comment:
just wondering - any reason this needs to be iterated in reverse order?

Reply:
Otherwise the indexes would be wrong...

Consider removing elements at 0, 1 from [a,b,c] of you iterate forwards `0` would remove `a`, leaving [b,c], then `1` would remove `c`, leaving [b]. Oops.